python/mxnet/random.py

```diff
@@ -38,7 +38,13 @@ def uniform(low, high, shape=None, ctx=None, out=None):
         if isinstance(shape, int):
             shape = (shape,)
         out = empty(shape, ctx)
-    return NDArray._sample_uniform(low=low, high=high, shape=out.shape, out=out)
+    if out.dtype == np.float32 or out.dtype ==np.float64:
+        return NDArray._sample_uniform(low=low, high=high, shape=out.shape, out=out)
+    else:
+        tmp = empty(out.shape, out.ctx)
+        tmp = NDArray._sample_uniform(low=low, high=high, shape=out.shape, out=tmp)
+        out[:] = tmp
+        return out
```

Comment:
only if out was not created before.. The out was already allocated in init_param function with Type. and here tmp is float type array.